pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -143,12 +143,27 @@ cdef inline _Timestamp create_timestamp_from_ts(
     """ convenience routine to construct a Timestamp from its parts """
     cdef:
         _Timestamp ts_base
-
-    ts_base = _Timestamp.__new__(Timestamp, dts.year, dts.month,
+        int64_t pass_year = dts.year
+
+    # We pass year=1970/1972 here and set year below because with non-nanosecond
```

Comment:
Could you reference why 1970/1972 is the base year? (guessing some Python implementation)

Reply:
If you could follow up with documenting the base year, that would be great